test/integration/services/modal.js

```diff
@@ -0,0 +1,144 @@
+  const smallOnTall = () => {
+    cy
+      .getRadio(Radio => {
+        Radio.request('modal', 'show:tall', {
+          headingText: 'Tall Modal Header',
+          bodyText: 'Tall Modal, which is under the small modal.',
+        });
+      });
+
+    cy
+      .getRadio(Radio => {
+        Radio.request('modal', 'show:small', {
+          headingText: 'Small Modal Header',
+          bodyText: 'Small Modal, this guy shows over the top of the tall modal.',
+        });
+      });
+  };
+  specify('Displaying', function() {
+    cy
+      .visit('/');
+
+    smallOnTall();
+
+    cy
+      .get('.modal--small')
+      .within(() => {
+        cy
+          .contains('Small Modal')
+          .get('.modal-cancel')
+          .click();
+      })
+      .get('.modal--tall')
+      .within(() => {
+        cy
+          .contains('Tall Modal')
+          .get('.modal-cancel')
+          .click();
+      });
+
+    // Close using the X icon
+    cy
+      .getRadio(Radio => {
+        Radio.request('modal', 'show:small', {
+          headingText: 'Small Modal Header',
+          bodyText: 'Small Modal',
+        });
+      });
+
+    cy
+      .get('.modal--small')
+      .find('.modal-close')
+      .click();
+
+    cy
+      .getRadio(Radio => {
+        const modal = Radio.request('modal', 'show:small', {
+          headingText: 'Close On Save',
+          bodyText: 'There\'s a callback to close the modal on save. Also a cool custom class!',
+          buttonClass: 'save-button',
+          className: 'modal--small custom-class',
+          onSubmit: () => {
+            modal.destroy();
+          },
+        });
+      });
+
+    cy
+      .get('.modal--small')
+      .then((modal) => {
+        expect(modal).to.have.class('custom-class');
+      })
+      .find('.save-button')
+      .click()
+      .should('not.exist');
+
+    cy
+      .getRadio(Radio => {
+        Radio.request('modal', 'show', {
+          footerView: false,
+          bodyText: 'No 🦶',
+        });
+      });
+
+    cy
+      .get('.modal')
+      .then((modal) => {
+        expect(modal).to.not.have.class('modal-footer');
+      });
+
+    cy
+      .getRadio(Radio => {
+        const modal = Radio.request('modal', 'show', {
+          bodyText: 'Submit Disabled',
+          onSubmit: () => {
+            modal.disableSubmit(true);
+          },
+        });
+      });
+
+    cy
+      .get('.modal')
+      .find('.js-submit')
+      .click()
+      .should('be.disabled');
+
+    smallOnTall();
+
+    // click the overlay, tall should stll be there
+    cy
+      .get('.fill-window--dark')
+      .first()
+      .next()
+      .click('left')
+      .get('.modal--tall')
+      .contains('Tall Modal')
+      .get('.fill-window--dark')
+      .click('left');
+
+    cy
+      .getRadio(Radio => {
+        const modal = Radio.request('modal', 'show', {
+          bodyText: 'Hi',
+        });
+        modal.getRegion('body').startPreloader();
```

Comment:
We never had this before as it was added later, but why don't we add a sugar function to the modal so that this can just be `modal.startPreloader()`